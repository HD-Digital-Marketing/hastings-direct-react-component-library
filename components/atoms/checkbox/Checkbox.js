import React from 'react'
import classNames from 'classnames'
import PropTypes from 'prop-types'
import './checkbox.scss'

const Checkbox = props => {

    const {
        className = '',
        theme = '',
        name,
        value,
        checked = false,
        disabled = false,
        readOnly = false,
        children
    } = props

    const checkboxClass = classNames('a-checkbox', className, {
        'a-checkbox--on-dark': theme === 'dark'
    })

    return (
        <div className={checkboxClass}>
            <input type="checkbox" id={name} name={name} value={value} defaultChecked={checked} disabled={disabled} readOnly={readOnly}/>
            <label>
                {children}
            </label>
        </div>
    )
}

Checkbox.propTypes = {
    type: PropTypes.string,
    className: PropTypes.string,
    theme: PropTypes.string,
    name: PropTypes.string,
    value: PropTypes.string,
    checked: PropTypes.bool,
    disabled: PropTypes.bool,
    readOnly: PropTypes.bool,
    children: PropTypes.string,
}

export default Checkbox
