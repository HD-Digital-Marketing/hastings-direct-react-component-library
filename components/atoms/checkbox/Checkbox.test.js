import React from 'react';
import renderer from 'react-test-renderer';
import Checkbox from './Checkbox';

describe('Check the checkbox renders', () => {
    test('it should match the snapshot', () => {

        const tree = renderer
            .create( <Checkbox id="checkboxOne" name="checkboxOne" value="Car" checked={true}>Car</Checkbox>)
            .toJSON();
        expect(tree).toMatchSnapshot();

    });
});