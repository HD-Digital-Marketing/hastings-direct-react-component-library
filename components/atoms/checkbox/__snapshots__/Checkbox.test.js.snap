// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check the checkbox renders it should match the snapshot 1`] = `
<div
  className="a-checkbox"
>
  <input
    defaultChecked={true}
    disabled={false}
    id="checkboxOne"
    name="checkboxOne"
    readOnly={false}
    type="checkbox"
    value="Car"
  />
  <label
    htmlFor="checkboxOne"
  >
    Car
  </label>
</div>
`;
